src/rtctools_heat_network/workflows/grow_workflow.py

```diff
@@ -523,18 +539,58 @@ def run_end_scenario_sizing(end_scenario_problem_class, staged_pipe_optimization
     if staged_pipe_optimization:
         solution = run_optimization_problem(end_scenario_problem_class, stage=1)
         results = solution.extract_results()
+        parameters = solution.parameters(0)
 
         # We give bounds for stage 2 by allowing one DN sizes larger than what was found in the
         # stage 1 optimization.
         pc_map = solution._HeatMixin__pipe_topo_pipe_class_map
         for pipe_classes in pc_map.values():
             v_prev = 0.0
+            first_pipe_class = True
             for var_name in pipe_classes.values():
                 v = results[var_name][0]
-                boolean_bounds[var_name] = (0.0, abs(v))
-                if v_prev == 1.0:
+                if first_pipe_class and abs(v) == 1.0:
+                    boolean_bounds[var_name] = (abs(v), abs(v))
```

Comment:
In theory I agree, but for now to create a faster running workflow in highs I think this worth it. The scenario you describe is possible but very unlikely to happen.

Reply:
I don't agree, because now you're than also not allowing to install additional producers/storages. Then the next booleans you fix for source and buffer placement you could also just fix that their boolean is 0 for stage 2 if not placed in stage 1.

Reply:
For now we'll leave it like this, because not using the first_pipe_class slows highs down by another 10 minutes